src/components/Filters/Views.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import {
+	DropDownButton,
+	Tooltip,
+	Material,
+	DropDownButtonProps,
+} from '@balena/ui-shared-components';
+import { faChartPie } from '@fortawesome/free-solid-svg-icons/faChartPie';
+import { JSONSchema7 as JSONSchema } from 'json-schema';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTrash } from '@fortawesome/free-solid-svg-icons/faTrash';
+import { useTranslation } from '../../hooks/useTranslation';
+import { Typography } from '@mui/material';
+
+const { Box, IconButton, useTheme, useMediaQuery } = Material;
+
+interface FiltersView {
+	id: string;
+	eventName: string;
+	name: string;
+	filters: JSONSchema[];
+}
+
+export interface ViewsProps {
+	views: FiltersView[] | undefined;
+	setFilters: (filters: JSONSchema[]) => void;
+	deleteView: (view: FiltersView) => any;
+	dark?: boolean;
+}
+
+export const Views = ({ views, setFilters, deleteView, dark }: ViewsProps) => {
+	const { t } = useTranslation();
+	const theme = useTheme();
+	const matches = useMediaQuery(theme.breakpoints.up('sm'));
+
+	const memoizedViews = React.useMemo<
+		DropDownButtonProps<FiltersView>['items'] | undefined
+	>(() => {
+		return views?.map((view) => {
+			return {
+				...view,
+				eventName: `Open saved view`,
+				children: (
+					<Box
+						display="flex"
+						alignItems="center"
+						justifyContent="space-between"
+						minWidth="200px"
+					>
+						<Typography>{view.name}</Typography>
+						<IconButton
+							aria-label={t('aria_labels.remove_view')}
+							onClick={() => deleteView(view)}
+						>
+							<FontAwesomeIcon icon={faTrash} />
+						</IconButton>
+					</Box>
+				),
+				onClick: () => {
+					const filters = view.filters;
+					setFilters(filters);
+				},
+			};
+		});
+	}, [views]);
+
+	return (
+		<Tooltip
+			title={!memoizedViews?.length ? t('labels.info_no_views') : undefined}
+		>
+			<DropDownButton<FiltersView>
+				disabled={!memoizedViews?.length}
+				items={memoizedViews ?? []}
+				children={
+					matches ? t('labels.views') : <FontAwesomeIcon icon={faChartPie} />
+				}
+				variant={!memoizedViews?.length && dark ? 'contained' : 'outlined'}
+				color={!memoizedViews?.length && dark ? 'primary' : 'secondary'}
```

Comment:
I think we can remove `dark` for now until we officially introduce dark theme

Reply:
this is mainly needed for filters in DeviceLogs, normal filters are not visible at all without dark. We can for sure discuss to improve it, I also hate this property, we need to remove it with a proper theme switch. Problem is that it is not easy to have 2 different filters in the same theme

Reply:
Ah okay. Another item for the TODO list then 😅